src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -62,9 +62,22 @@ public partial class MatchCollection : System.Collections.ICollection, System.Co
         void System.Collections.ICollection.CopyTo(System.Array array, int arrayIndex) { }
     }
     public delegate string MatchEvaluator(System.Text.RegularExpressions.Match match);
+    abstract public partial class RegexRunnerFactory
+    {
+        protected RegexRunnerFactory() { }
+        abstract protected internal RegexRunner CreateInstance();
+    }
     public partial class Regex
     {
         public static readonly System.TimeSpan InfiniteMatchTimeout;
+        protected internal string pattern;
```

Comment:
The reason they are exposed is because the pre-compiled regex generator for desktop framework directly accesses them and we want that scenario to be able to work in .NET Core. We realize this isn't the greatest design but we have to balance that with making existing assets work.